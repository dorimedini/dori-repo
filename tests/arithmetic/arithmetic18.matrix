{
	int a = 2;
	int b = 5;
	int x;
	
	// Addition and subtraction have same precedence so the earlier should operate first
	x=a - b + a;
	print x;
	x=a + b - a;
	print x;

	// Multiplication and division go before addition and subtraction
	x=a - a*b;
	print x;
	x=b + b/b;
	print x;
	x=a * b / a;
	print x;
	x=a / a * b;
	print x;

	// Unary minus comes before arithmetic operations
	x=a + -b;
	print x;
	x=a + --b;
	print x;
	x=a + ---b;
	print x;
	x=b / -a;
	print x;
	x=b / --a;
	print x;
	x=b / ---a;
	print x;


	matrix m1(3,3) = [1, 2, 3; 4, 5, 6; 7, 8, 9];
	matrix m2(3,3) = [3, 3, 3; 2, 2, 2; 1, 1, 1];
	matrix m3(3,2) = [1, 2; 4, 5; 7, 8];
	matrix m4(3,1) = [3; 6; 9];
	matrix m33(3,3);

	m33=m1 - m2 + m1;
	print m33;
	m33=m1 + m2 - m1;
	print m33;

	m33=m1 - m1*m2;
	print m33;
	m33=m1 + m1*m2;
	print m33;

	m33=m1 + -m2;
	print m33;
	m33=m1 + --m2;
	print m33;
	m33=m1 + ---m2;
	print m33;
	m33=m1 * -m2;
	print m33;
	m33=m1 * --m2;
	print m33;
	m33=m1 * ---m2;
		
}