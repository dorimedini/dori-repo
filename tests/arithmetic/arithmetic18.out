-1
5
-8
6
5
5
-3
7
-3
-2
2
-2
[-1,1,3;6,8,10;13,15,17]
[3,3,3;2,2,2;1,1,1]
[-9,-8,-7;-24,-23,-22;-39,-38,-37]
[11,12,13;32,33,34;53,54,55]
[-2,-1,0;2,3,4;6,7,8]
[4,5,6;6,7,8;8,9,10]
[-2,-1,0;2,3,4;6,7,8]
[-10,-10,-10;-28,-28,-28;-46,-46,-46]
[10,10,10;28,28,28;46,46,46]
[-10,-10,-10;-28,-28,-28;-46,-46,-46]
